### ISSUE TITLE
Update Call notify_template with custom field

### ISSUE THREAD

**Issue by TeuZz** [salesagility/SuiteCRM#7785]
Hi,

I found a solution to update my "notify_template.html" with custom field, and I would like to suggest it to you.

#### Context
I have to update the Call notification with adding the "related to" name.
I saw I can write in "modules\Calls\Call.php" and add a line in "set_notification_body" : 

`$xtpl->assign("CALL_CONTACT", $call->parent_name);`

The problem is : "how do this change in custom part ?"

I tried to create a CustomCall class, with an override "set_notification_body" but it wasn't work.

After many investigation, i found the guilty ! 
CustomCall->save() is never call because the file "modules\Calls\CallFromBase.php" got a function "handleSave" who create a 
`$focus = new Call();`

#### Possible Fix
To take the customCall class, we need : 
`$focus = BeanFactory::getBean('Calls');`

#### Your Environment

* SuiteCRM Version used: 7.11.8

**Comment by pgorod** [comment]
@TeuZz if this is just a suggestion to help fellow coders, please make it on the forums:

https://suitecrm.com/suitecrm/forum/

Or is this something that should be changed in the core code? I am not sure I understand completely what this is about. If you can please clarify... thanks!

**Comment by TeuZz** [comment]
Hi @pgorod 

This is a suggestion to change the core code.

My problem was to change informations display in notification email for "Calls".
I had to add the contact (or account) name link with the call.
I did it in adding a variable in the notify_template file and assign the value in "modules\Calls\Call.php" with : 
`$xtpl->assign("CALL_CONTACT", $call->parent_name);`

But, I want to do this in custom part to be upgrade safe.
When I create a "custom\modules\Calls\Call.php", it's not working cause the custom class is not use.
The file "modules\Calls\CallFromBase.php" call functions for an object created by
`new Call()`
but, to use custom class, we need to create this object with 
`BeanFactory::getBean('Calls')`
With this way, we can create a custom "Calls" class.

**To recap :** 
If we change the file "modules\Calls\CallFromBase.php" from 
`new Call()` to `BeanFactory::getBean('Calls')`
we will be free to custom "Calls" module.

I hope that explains it a little better.
